### PR TITLE
Refactored to handle errors better with .then(onResolved, onRejected) and look out for 404

### PR DESCRIPTION
**Pull request by kaihendry**

### CODE REVIEW

File: aws-node-fetch-file-and-store-in-s3/handler.js [hunk 1]
```diff
@@ -2,24 +2,19 @@
...
 const AWS = require('aws-sdk');
-AWS.config.setPromisesDependency(require('bluebird'));
 
 const s3 = new AWS.S3();
 
 module.exports.save = (event, context, callback) => {
   fetch(event.image_url)
-    .then(response => response.buffer())
-    .then(buffer => (
+  .then(res => (res.ok ? res : Promise.reject({ status: res.status, text: res.statusText })))
```
Comment (christophgysin) [inline aws-node-fetch-file-and-store-in-s3/handler.js]: This is good! Could you make a pull request with this change only?

The other changes are just rearranging the code, shortening variable names, changing indentation or have already been merged.

Comment (kaihendry) [inline aws-node-fetch-file-and-store-in-s3/handler.js]: Apologies, I'm a bit of a newb with Github PRs. Am I correct in thinking I need to:

* reset my 'refactor' branch
* make this one line change
* commit it
* `git push -f`

Is that right? Surely it's easier for you just to cherry pick the line?!

Also the more important aspect I was trying to address is that callback maybe called unpredictably from two places: https://github.com/serverless/examples/blob/master/aws-node-fetch-file-and-store-in-s3/handler.js#L19-L22

Comment (christophgysin) [inline aws-node-fetch-file-and-store-in-s3/handler.js]: Ah, I missed the second change. My point was to try and reduce noise (indentation, renaming, etc.) so it becomes more clear what this change is doing. Also, consider putting that last explanation into to commit body.

Regarding losing your changes with force push: You can either make a separate commit on top, or store your original changes in another branch. If you already force pushed you can still get your old change back from the reflog.

Comment (kaihendry) [inline aws-node-fetch-file-and-store-in-s3/handler.js]: Ok, I was trying to explain what I was going in the subject of the PR. Oops. :grin:

**Comment by kaihendry** [comment]
ARGH, sorry, I just tested the change and I realise my line should be after L9, not L10... APOLOGIES

**Comment by christophgysin** [comment]
@kaihendry Could you check #124?

**Pull request merged**